scripts/set_compiler_env.sh

```diff
@@ -1,84 +1,54 @@
 #!/bin/sh
+set -eu
 
-# Script to configure environment variables for Go compiler
-# to allow cross compilation
-
-: ${TARGETPLATFORM=}
-: ${TARGETOS=}
-: ${TARGETARCH=}
-: ${TARGETVARIANT=}
+# Configure environment for cross-compiling.
 
-CGO_ENABLED="$(go env CGO_ENABLED)"
-GOARCH="$(go env GOARCH)"
-GOOS="$(go env GOOS)"
-GOARM="$(go env GOARM)"
-GOBIN="$(go env GOBIN)"
+: ${TARGETPLATFORM=linux/`dpkg --print-architecture`}
 
-set -eu
+TARGETOS="$(echo $TARGETPLATFORM | cut -d"/" -f1)"
+TARGETARCH="$(echo $TARGETPLATFORM | cut -d"/" -f2)"
+TARGETVARIANT="$(echo $TARGETPLATFORM | cut -d"/" -f3)"
 
-if [ ! -z "$TARGETPLATFORM" ]; then
-  TARGETOS="$(echo $TARGETPLATFORM | cut -d"/" -f1)"
-  TARGETARCH="$(echo $TARGETPLATFORM | cut -d"/" -f2)"
-  TARGETVARIANT="$(echo $TARGETPLATFORM | cut -d"/" -f3)"
+DEBIAN_ARCH=$TARGETARCH
+if [ "$TARGETARCH" = "arm" ]
+then
+  DEBIAN_ARCH=armel
+  if [ "$TARGETVARIANT" = "v7" ]
+  then
+    DEBIAN_ARCH=armhf
+  fi
 fi
 
-if [ ! -z "$TARGETOS" ]; then
-  export GOOS="$TARGETOS"
-fi
+dpkg --add-architecture $DEBIAN_ARCH
+apt-get update
+apt-get install -y git curl crossbuild-essential-${DEBIAN_ARCH} libc-dev:${DEBIAN_ARCH} autoconf automake libtool m4 pkg-config cmake
 
-if [ ! -z "$TARGETARCH" ]; then
-  export GOARCH="$TARGETARCH"
-fi
+dpkg-architecture -a $DEBIAN_ARCH >/env
+export $(cat /env)
 
-if [ "$TARGETARCH" = "arm" ]; then
-  if [ ! -z "$TARGETVARIANT" ]; then
-    case "$TARGETVARIANT" in
-    "v5")
-      export GOARM="5"
-      ;;
-    "v6")
-      export GOARM="6"
-      ;;
-    *)
-      export GOARM="7"
-      ;;
-    esac
-  else
-    export GOARM="7"
-  fi
-fi
+CGO_ENABLED="1"
+GOOS="$TARGETOS"
+GOARCH="$TARGETARCH"
 
-if [ "$CGO_ENABLED" = "1" ]; then
-  case "$GOARCH" in
-  "amd64")
-    export COMPILER_ARCH="x86_64-linux-gnu"
-    ;;
-  "ppc64le")
-    export COMPILER_ARCH="powerpc64le-linux-gnu"
+GOARM="7"
+if [ "$TARGETARCH" = "arm" && ! -z "$TARGETVARIANT" ]; then
```

Comment:
I guess that you forgot to update this code and remove unsupported arch's

Reply:
See [the previous thread](https://github.com/photoview/photoview/pull/1053#discussion_r1761715575).